### PR TITLE
[#5310] add management command to calculate bag sizes for "stale" resources

### LINKED ISSUE #5310: If no downloads within a year, should we "offboard" a resource's zipped bagit archive
As we contemplate changing our storage backend and aim to turn on quota enforcement, it would be good to contemplate how much of our storage is being used by "cold" files that no one is accessing. Zipped bagit archives could be removed after some interval of inactivity (I propose a year)--in this case "inactivity" would mean no recent downloads.

### PR DESCRIPTION
**Pull request by devincowan**
Hoping might eventually fixes #5310 

### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

File: hs_core/management/commands/compute_bag_size_depending_on_last_update.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        s = round(size_bytes / p, 2)
+        return "%s %s" % (s, size_name[i])
+
+    def handle(self, *args, **options):
+        resources = []
+        add_message = ""
+        months = options['months']
+        if len(options['resource_ids']) > 0:  # an array of resource short_id to check.
+            if months:
+                raise CommandError("Cant privide resource_ids and also filter by months")
```
Comment (sblack-usu) [inline hs_core/management/commands/compute_bag_size_depending_on_last_update.py:36]: Is there a use case to use this management command with the resources_ids argument?  Can the resource_ids argument be removed?

File: hs_core/management/commands/compute_bag_size_depending_on_last_update.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            for rid in options['resource_ids']:
+                resources.append(get_resource_by_shortkey(rid))
+        else:
+            resources = BaseResource.objects.all()
+
+        if months:
+            print(f"FILTERING TO INCLUDE RESOURCES NOT UPDATED IN UPDATED IN LAST {months} MONTHS")
+            add_message = f"for resources not updated in last {months} months"
+            cuttoff_time = timezone.now() - timedelta(months)
+            resources = resources.filter(updated__lte=cuttoff_time)
```
Comment (sblack-usu) [inline hs_core/management/commands/compute_bag_size_depending_on_last_update.py:46]: is the updated time the correct field to check?  It'd be nice to know when the last download time was, but we're not tracking that.  Updated may be the best we have.  I'm just concerned it doesn't capture the information we are looking for.  What if there is a resource that hasn't changed in years but is downloaded weekly?

Comment (devincowan) [reply hs_core/management/commands/compute_bag_size_depending_on_last_update.py:46]: you're 100% spot on. I agree, see https://github.com/hydroshare/hydroshare/issues/5310#issuecomment-1908938806

I think this is the best we can do for now. My thought is that we deploy this command, run it to get a general feel for how big our "bag" storage is. If it is large enough that it warrants adding a `last_downloaded_date` then we add that and start tracking it.

File: hs_core/management/commands/compute_bag_size_depending_on_last_update.py (hunk 1 above)
Comment (devincowan) [reply hs_core/management/commands/compute_bag_size_depending_on_last_update.py]: good call, probably not. Removed it

**Review by sblack-usu — APPROVED** [review]

**Pull request merged**